h/views/account_signup.py

```diff
@@ -257,6 +324,59 @@ def post(self):
             headers=login(user, self.request),
         )
 
+    @exception_view_config(
+        context=EmailConflictError,
+        route_name="signup.orcid",
+        renderer="h:templates/error.html.jinja2",
+    )
+    @exception_view_config(
+        context=EmailConflictError,
+        route_name="signup.google",
+        renderer="h:templates/error.html.jinja2",
+    )
+    @exception_view_config(
+        context=EmailConflictError,
+        route_name="signup.facebook",
+        renderer="h:templates/error.html.jinja2",
+    )
+    def email_conflict_error(self):
+        self.request.session.flash(
+            f"There's already a Hypothesis account with your {self.settings.provider_name} email address."
+            " Try logging in or resetting your password."
+            f" Once logged in you can connect {self.settings.provider_name} in your account settings.",
+            "error",
+        )
+
+        return HTTPFound(
+            self.request.route_url(
+                "login", _query={"username": self.decode_idinfo().email}
+            ),
+            headers=logout(self.request),
+        )
```

Comment:
This exception view called if either the `GET` or `POST` request to `/signup/<PROVIDER>` raises `EmailConflictError`. We redirect to the login page with a flash error message.

Note that we also log the user out when doing this (in case during the signup flow they had logged in in another tab), to prevent the login page from immediately redirecting to the user's profile page with potentially confusing results (especially when the logged-in account may not match the one the user just tried to signup).